app/controllers/declaration/BorderTransportController.scala

```diff
@@ -40,24 +41,34 @@ class BorderTransportController @Inject()(
   customsCacheService: CustomsCacheService,
   mcc: MessagesControllerComponents
 )(implicit ec: ExecutionContext, appConfig: AppConfig)
-    extends FrontendController(mcc) with I18nSupport {
+  extends FrontendController(mcc) with I18nSupport {
 
   def displayForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+
     customsCacheService
       .fetchAndGetEntry[BorderTransport](cacheId, BorderTransport.formId)
-      .map(data => Ok(border_transport(data.fold(form)(form.fill(_)))))
+      .flatMap { borderTransportData =>
+        customsCacheService
+          .fetchAndGetEntry[FiscalInformation](cacheId, FiscalInformation.formId)
+          .map(data => Ok(border_transport(borderTransportData.fold(form)(form.fill(_)), data.fold(false)(_.onwardSupplyRelief == "Yes"))))
+      }
   }
 
-  def submitForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    form
-      .bindFromRequest()
-      .fold(
-        (formWithErrors: Form[BorderTransport]) => Future.successful(BadRequest(border_transport(formWithErrors))),
-        borderTransport =>
-          customsCacheService
-            .cache[BorderTransport](cacheId, BorderTransport.formId, borderTransport)
-            .map(_ => Redirect(TransportDetailsController.displayForm()))
-      )
-  }
+    def submitForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          (formWithErrors: Form[BorderTransport]) => badRequest(formWithErrors),
+          borderTransport =>
+            customsCacheService
+              .cache[BorderTransport](cacheId, BorderTransport.formId, borderTransport)
+              .map(_ => Redirect(TransportDetailsController.displayForm()))
+        )
+    }
+
+    def badRequest(formWithErrors: Form[BorderTransport])(implicit request: JourneyRequest[_], appConfig: AppConfig) =
```

Comment:
I think that instead of extracting `badRequest` method (it used in only one place) you can extract method to generate view based on the form. So you will be able to reuse it in `displayForm` method too.

Reply:
Addressed! :)